src/shims/unix/wasi/foreign_items.rs

```diff
@@ -0,0 +1,187 @@
+use std::ffi::OsString;
+
+use rustc_span::Symbol;
+use rustc_target::abi::Size;
+use rustc_target::spec::abi::Abi;
+
+use crate::*;
+use shims::foreign_items::EmulateByNameResult;
+
+use rustc_middle::ty::layout::LayoutOf as _;
+use rustc_middle::ty::Ty;
+
+impl<'mir, 'tcx: 'mir> EvalContextExt<'mir, 'tcx> for crate::MiriInterpCx<'mir, 'tcx> {}
+pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriInterpCxExt<'mir, 'tcx> {
+    fn emulate_foreign_item_by_name(
+        &mut self,
+        link_name: Symbol,
+        abi: Abi,
+        args: &[OpTy<'tcx, Provenance>],
+        dest: &PlaceTy<'tcx, Provenance>,
+    ) -> InterpResult<'tcx, EmulateByNameResult<'mir, 'tcx>> {
+        let this = self.eval_context_mut();
+
+        // See `fn emulate_foreign_item_by_name` in `shims/foreign_items.rs` for the general pattern.
+        match link_name.as_str() {
+            "args_sizes_get" => {
```

Comment:
add a comment explaining that this writes the number of arguments and the size (including null terminators) to the passed pointers, maybe with a link to docs about this, if there are any